### PR TITLE
tests: allow platform specific unit test files

### PR DESCRIPTION
**Pull request by ReenigneArcher**
## Description

Allow platform specific test code, other than the platform specific fixtures.

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [x] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I want maintainers to keep my branch updated

### CODE REVIEW

**Comment by ReenigneArcher** [comment]
@FrogTheFrog FYI.

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -29,6 +29,22 @@ file(GLOB_RECURSE TEST_SOURCES
         ${CMAKE_SOURCE_DIR}/tests/conftest.cpp
         ${CMAKE_SOURCE_DIR}/tests/utils.cpp
```
Comment (FrogTheFrog) [inline tests/CMakeLists.txt:30]: Could we also glob for any `.h, .cpp` to be added instead (unless you want to separate `test_*.cpp` somehow to enforce test name pattern)?

Same for `${CMAKE_SOURCE_DIR}/tests/*/platf/windows/` level. I know that I will have some mocks in `${CMAKE_SOURCE_DIR}/tests/*/platf/windows/` that will belong to windows only.

Comment (ReenigneArcher) [reply tests/CMakeLists.txt:30]: I don't think `.h` is needed here. I know we do that in Sunshine, but I think that was a mistake. It should be covered by `include_directories("${CMAKE_SOURCE_DIR}")`

I would like to enforce a naming pattern just for consistency, but they don't necessarily all need to start with `test_`. Is there a pattern you are predicting to use? Maybe `setup_` or `mock_`?

Comment (FrogTheFrog) [reply tests/CMakeLists.txt:30]: I would be fine with `mock_`.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -29,6 +29,22 @@ file(GLOB_RECURSE TEST_SOURCES
...
+            ${CMAKE_SOURCE_DIR}/tests/*/platf/windows/test_*.cpp)
+elseif(__APPLE__)
+    file(GLOB_RECURSE TEST_PLATFORM_SOURCES
+            ${CMAKE_SOURCE_DIR}/tests/*/platf/macos/test_*.cpp)
+elseif(UNIX)
+    file(GLOB_RECURSE TEST_PLATFORM_SOURCES
+            ${CMAKE_SOURCE_DIR}/tests/*/platf/linux/test_*.cpp)
+else()
+    message(FATAL_ERROR "Unsupported platform")
+endif()
```
Comment (ReenigneArcher) [inline tests/CMakeLists.txt:46]:
```suggestion
if(WIN32)
    file(GLOB_RECURSE TEST_PLATFORM_SOURCES
            "${CMAKE_SOURCE_DIR}/tests/*/platf/windows/test_*.cpp")
elseif(__APPLE__)
    file(GLOB_RECURSE TEST_PLATFORM_SOURCES
            "${CMAKE_SOURCE_DIR}/tests/*/platf/macos/test_*.cpp")
elseif(UNIX)
    file(GLOB_RECURSE TEST_PLATFORM_SOURCES
            "${CMAKE_SOURCE_DIR}/tests/*/platf/linux/test_*.cpp")
else()
    message(FATAL_ERROR "Unsupported platform")
endif()
```

File: tests/CMakeLists.txt (hunk 1 above)
Comment (ReenigneArcher) [inline tests/CMakeLists.txt:30]:
```suggestion
        "${CMAKE_SOURCE_DIR}/tests/conftest.cpp"
        "${CMAKE_SOURCE_DIR}/tests/utils.cpp"
```

**Pull request merged**